os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -77,6 +77,51 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+static void allocateregions(FAR struct elf_loadinfo_s *loadinfo)
+{
+	uint32_t totalsize;
+	size_t textsize, datasize, rosize;
+
+	textsize = loadinfo->textsize;
```

Comment:
If there is a explanation why we try this, then we don't need to declare the variable with each section name.
We can just declare size_t sections[3], sort it by decreasing order, and allocate from start to end, we can reduce the code.
How about think of my suggestion?

Reply:
If we use size_t sections[3], then after sorting them in order, then we will not know which array element is for which type of section. So, it will be difficult to perform the allocation assignments.